rust/perspective-viewer/src/rust/components/column_settings_sidebar.rs

```diff
@@ -0,0 +1,128 @@
+use crate::components::containers::sidebar::Sidebar;
+use crate::components::containers::tablist::TabList;
+use crate::components::style::LocalStyle;
+use crate::presentation::Presentation;
+use crate::renderer::Renderer;
+use crate::session::Session;
+use crate::{clone, css, html_template};
+
+#[derive(Debug, Default, Clone, PartialEq)]
+pub enum ColumnSettingsTab {
+    #[default]
+    Attributes,
+    Style,
+}
+impl Display for ColumnSettingsTab {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!("{self:?}"))
+    }
+}
+impl Tab for ColumnSettingsTab {}
+
+#[derive(PartialEq, Clone, Properties)]
+pub struct ColumnSettingsProps {
+    pub selected_column: ColumnLocator,
+    pub on_close: Callback<()>,
+    pub session: Session,
+    pub renderer: Renderer,
+    pub presentation: Presentation,
+}
+
+#[function_component]
+pub fn ColumnSettingsSidebar(p: &ColumnSettingsProps) -> Html {
+    let column_name = match p.selected_column.clone() {
+        ColumnLocator::Expr(maybe_name) => maybe_name.unwrap_or("New Column".to_string()),
+        ColumnLocator::Plain(name) => name,
+    };
+
+    let column_type = p.session.metadata().get_column_view_type(&column_name);
+    let is_active = column_type.is_some();
+
+    let mut tabs = vec![];
+
+    // Eventually the Attributes tab will have more properties than
+    // just the expression editor. Once that happens, stop hiding it.
+    if matches!(p.selected_column, ColumnLocator::Expr(_)) {
+        tabs.push(ColumnSettingsTab::Attributes);
+    }
+    if !matches!(p.selected_column, ColumnLocator::Expr(None)) && is_active {
+        tabs.push(ColumnSettingsTab::Style);
+    }
+
+    let title = format!("Editing ‘{column_name}’...");
+
+    clone!(
+        p.selected_column,
+        p.on_close,
+        p.session,
+        p.renderer,
+        p.presentation,
+        column_name
+    );
+    let match_fn = Callback::from(move |tab| {
+        clone!(
+            selected_column,
+            on_close,
+            session,
+            renderer,
+            presentation,
+            column_name
+        );
+        match tab {
+            ColumnSettingsTab::Attributes => html! {
+                <AttributesTab
+                    { selected_column }
+                    { on_close }
+                    { session }
+                    { renderer }
+                />
+            },
+            ColumnSettingsTab::Style => html! {
+                <StyleTab
+                    { column_name }
+                    { session }
+                    { renderer }
+                    { presentation }
+                />
+            },
+        }
+    });
+
+    html_template! {
+        <LocalStyle href={css!("column-settings-panel")} />
+        <Sidebar
+            {title}
+            on_close={p.on_close.clone()}
+            id_prefix="column_settings"
+            icon={html_template!{
```

Comment:
IMO this is overkill - the icon data is metadata that can be passed as a prop, it's never going to be some other DOM structure, in this case the `id` attribute.